### PR TITLE
Add Azure search E2E test to verify owner information

### PR DESCRIPTION
**Pull request by shishirx34**
Addresses: https://github.com/NuGet/NuGetGallery/issues/6464 

There is also a new config added for "TestAccountOwner", will send the PR shortly. Verified this test in DEV.

### CODE REVIEW

File: src/NuGet.Services.EndToEnd/PushTests.cs [hunk 1]
```diff
@@ -106,5 +108,30 @@ public async Task NewlyPushedIsODataSearchableInDB(PackageType packageType)
...
+            var searchServices = await _clients.V2V3Search.GetSearchServicesAsync(_logger);
+
+            // wait for package to become available
+            await _clients.V2V3Search.WaitForPackageAsync(package.Id, package.FullVersion, _logger);
+
+            // Act
+            foreach (var searchService in searchServices)
+            {
+                var result = await _clients.V2V3Search.QueryAsync(searchService, $"q=packageid:{package.Id} owner:{_testSettings.TestAccountOwner}", _logger);
+                Assert.NotNull(result);
```
Comment (joelverhagen) [inline src/NuGet.Services.EndToEnd/PushTests.cs]: nit `// Assert` before the assert lines

File: src/NuGet.Services.EndToEnd/PushTests.cs [hunk 2]
```diff
@@ -106,5 +108,30 @@ public async Task NewlyPushedIsODataSearchableInDB(PackageType packageType)
...
+                Assert.NotNull(result.Data);
+                Assert.NotEmpty(result.Data);
+                Assert.Equal(1, result.Data.Count);
```
Comment (joelverhagen) [inline src/NuGet.Services.EndToEnd/PushTests.cs]: Nit: `Assert.Single`

File: src/NuGet.Services.EndToEnd/Support/TestSettings.cs [hunk 3]
```diff
@@ -56,6 +58,12 @@ private TestSettings()
         /// </summary>
         public bool SkipGalleryTests { get; set; }
 
+        public bool IsTestingAzureSearchService()
```
Comment (joelverhagen) [inline src/NuGet.Services.EndToEnd/Support/TestSettings.cs]: Can you create an issue to clean this fact up later when we are off old search?

File: src/NuGet.Services.EndToEnd/PushTests.cs [hunk 4]
```diff
@@ -106,5 +108,30 @@ public async Task NewlyPushedIsODataSearchableInDB(PackageType packageType)
...
+            // wait for package to become available
+            await _clients.V2V3Search.WaitForPackageAsync(package.Id, package.FullVersion, _logger);
+
+            // Act
+            foreach (var searchService in searchServices)
+            {
+                var result = await _clients.V2V3Search.QueryAsync(searchService, $"q=packageid:{package.Id} owner:{_testSettings.TestAccountOwner}", _logger);
+                Assert.NotNull(result);
+                Assert.NotNull(result.Data);
+                Assert.NotEmpty(result.Data);
```
Comment (joelverhagen) [inline src/NuGet.Services.EndToEnd/PushTests.cs]: `NotEmpty` and the count assertion are redundant. I recommend just removing this one.

File: src/NuGet.Services.EndToEnd/Support/TestSettings.cs (hunk 3 above)
Comment (shishirx34) [reply src/NuGet.Services.EndToEnd/Support/TestSettings.cs]: https://github.com/NuGet/Engineering/issues/2636

**Pull request merged**